batch/batch/worker/worker.py

```diff
@@ -419,135 +489,232 @@ async def batch_worker_access_token(self):
-                self.state = 'failed'
-        except asyncio.CancelledError:
-            raise
-        except Exception as e:
-            if not isinstance(e, (JobDeletedError, JobTimeoutError)):
-                log.exception(f'while running {self}')
+    def write_container_config(self):
+        os.makedirs(self.config_path)
+        with open(f'{self.config_path}/config.json', 'w') as f:
+            f.write(json.dumps(self.container_config()))
 
-            self.state = 'error'
-            self.error = traceback.format_exc()
-        finally:
-            await self.delete_container()
+    # https://github.com/opencontainers/runtime-spec/blob/master/config.md
+    def container_config(self):
+        weight = worker_fraction_in_1024ths(self.spec['cpu'])
+        default_docker_capabilities = [
+            'CAP_CHOWN',
+            'CAP_DAC_OVERRIDE',
+            'CAP_FSETID',
+            'CAP_FOWNER',
+            'CAP_MKNOD',
+            'CAP_NET_RAW',
+            'CAP_SETGID',
+            'CAP_SETUID',
+            'CAP_SETFCAP',
+            'CAP_SETPCAP',
+            'CAP_NET_BIND_SERVICE',
+            'CAP_SYS_CHROOT',
+            'CAP_KILL',
+            'CAP_AUDIT_WRITE',
+        ]
+        return {
+            'ociVersion': '1.0.1',
+            'root': {
+                'path': '.',
+                'readonly': False,
+            },
+            'hostname': self.netns.hostname,
+            'mounts': self._mounts(),
+            'process': {
+                'user': {  # uid/gid *inside the container*
+                    'uid': 0,
+                    'gid': 0,
+                },
+                'args': self.spec['command'],
+                'env': self._env(),
+                'cwd': '/',
+                'capabilities': {
+                    'bounding': default_docker_capabilities,
+                    'effective': default_docker_capabilities,
+                    'inheritable': default_docker_capabilities,
+                    'permitted': default_docker_capabilities,
+                },
+            },
+            'linux': {
+                'namespaces': [
+                    {'type': 'pid'},
+                    {
+                        'type': 'network',
+                        'path': f'/var/run/netns/{self.netns.network_name}',
+                    },
+                    {'type': 'mount'},
+                    {'type': 'ipc'},
+                    {'type': 'uts'},
+                    {'type': 'cgroup'},
+                ],
+                'uidMappings': [],  # Can control host -> container uid/gid mappings here
+                'gidMappings': [],
+                'resources': {
+                    'cpu': {'shares': weight},
+                    'memory': {
+                        'limit': self.spec['memory'],
+                        'reservation': self.spec['memory'],
+                    },
+                    # 'blockIO': {'weight': min(weight, 1000)}, TODO Got permission denied
```

Comment:
Interestingly, linux 5.0 removed `blkio.weight`/`blkio.weight_device` and in our production logs right now you see a lot of
`Your kernel does not support cgroup blkio weight` and `Your kernel does not support cgroup blkio weight_device` warnings from `dockerd`, so I'm inclined to leave this as a `FIXME` and come back to it in a separate PR.